v0/destinations/mailmodo/transform.js

```diff
@@ -0,0 +1,297 @@
+    // Batch event into dest batch structure
+    chunk.forEach(ev => {
+      values.push(ev?.message?.body?.JSON?.values[0]);
+      metadatas.push(ev.metadata);
+    });
+    // batching into identify batch structure
+    batchEventResponse.batchedRequest.body.JSON = {
+      listName: `${listName}`,
+      values
+    };
+    batchEventResponse.batchedRequest.endpoint = `${BASE_URL}/addToList/batch`;
+
+    batchEventResponse.batchedRequest.headers = {
+      "Content-Type": "application/json",
+      mmApiKey: apiKey
+    };
+
+    batchEventResponse = {
+      ...batchEventResponse,
+      metadata: metadatas,
+      destination
+    };
+    batchedResponseList.push(
+      getSuccessRespEvents(
+        batchEventResponse.batchedRequest,
+        batchEventResponse.metadata,
+        batchEventResponse.destination,
+        true
+      )
+    );
+  });
+
+  return batchedResponseList;
+}
+
+function getEventChunks(event, identifyEventChunks, eventResponseList) {
+  // Categorizing identify and track type of events
+  // Checking if it is identify type event
+  if (event.message.endpoint.includes("/addToList/batch")) {
+    identifyEventChunks.push(event);
+  } else {
+    // any other type of event
+    const { message, metadata, destination } = event;
+    const endpoint = get(message, "endpoint");
+
+    const batchedResponse = defaultBatchRequestConfig();
+    batchedResponse.batchedRequest.headers = message.headers;
+    batchedResponse.batchedRequest.endpoint = endpoint;
+    batchedResponse.batchedRequest.body = message.body;
+    batchedResponse.batchedRequest.params = message.params;
+    batchedResponse.batchedRequest.method =
+      defaultPostRequestConfig.requestMethod;
+    batchedResponse.metadata = [metadata];
+    batchedResponse.destination = destination;
+
+    eventResponseList.push(
+      getSuccessRespEvents(
+        batchedResponse.batchedRequest,
+        batchedResponse.metadata,
+        batchedResponse.destination
+      )
+    );
+  }
+}
+
+const processRouterDest = inputs => {
+  if (!Array.isArray(inputs) || inputs.length <= 0) {
+    const respEvents = getErrorRespEvents(null, 400, "Invalid event array");
+    return [respEvents];
+  }
+
+  let identifyEventChunks = []; // list containing identify events in batched format
+  const eventResponseList = []; // list containing other events in batched format
+  const identifyArrayChunks = [];
+  const errorRespList = [];
+  Promise.all(
+    inputs.map((event, index) => {
+      try {
+        if (event.message.statusCode) {
+          // already transformed event
+          getEventChunks(event, identifyEventChunks, eventResponseList);
+          // slice according to batch size
+          if (
+            identifyEventChunks.length &&
+            (identifyEventChunks.length >= IDENTIFY_MAX_BATCH_SIZE ||
+              index === inputs.length - 1)
+          ) {
+            identifyArrayChunks.push(identifyEventChunks);
+            identifyEventChunks = [];
```

Comment:
This logic is hard to understand. Let's don't divide in chunks here. Put everything in array and break them in chunks later part.

Reply:
Modified accordingly